packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -20,7 +20,11 @@ const defaultGetCollection = () => null;
  * @return {Array} Words, extracted from the input string.
  */
 function extractWords( input = '' ) {
-	return noCase( input, {
+	if ( typeof extractWords[ input ] !== 'undefined' ) {
+		return extractWords[ input ];
```

Comment:
Ha, I never though of using the function itself as the lookup table! But I think this might be unsafe because the `input` can collide with existing keys like `length` or `apply` (it crashes in this implementation).

Reply:
Heh, that's fair. I considered those, but thought that since those functions are private and we have full control over their use, and we don't really use any of those function prototype features, we're safe. But you're right: if someone starts using them in the future, we have no reliable way to prevent from the bugs that may occur.

That being said, I've refactored to using dedicated objects in 6f16057d3bde2f28cf8e0aeafacc23fc9f9d3be5. Feel free to take another look!

Reply:
I'm mostly concerned about crashes due to the user searching for terms that collide with built-in keys. This empty object implementation decreases the surface area of the crashes, but can still crash on certain collisions (e.g. when you search for `"constructor"`). It would be safer (and probably more performant?) to use a `Map`.

Reply:
IMO, the set might be more suitable. Search terms will be unique strings.

Reply:
Good point, updated to use `Map` in f801b59cb286a2cb7711e79a22f17b1908bf0fba. 

A `Set` won't do, because we maintain them in a key-value manner, and `Set` is good only for unique value storage.